### DISCUSSION TITLE
In bash reverse-search, can I return the search query as-is?

### DISCUSSION THREAD

**Discussion by simonschmidt** [junegunn/fzf#3970] [Q&A] [answered]
Here's what I'm doing, with my shell set up with `fzf --bash`:

1. I want to run `foo --bar --baz`
2. Ctrl+R in bash to start reverse history search
3. Start typing, turns out I haven't ran `foo --bar --baz` before
4. End up with `foo --bar --baz` as my search query all typed out
5. Realize I can't select it
6. Ctrl+C to exit reverse search
7. Retype command

Is it possible to at any point abort the search returning current search-term as-is, perhaps a shortcut?

**Comment by junegunn** [comment]
See https://github.com/junegunn/fzf/issues/3528

FWIW, fzf is fuzzy finder, you don't have to type the exact command in the prompt. Your query can be "fuzzy". e.g. `fo-babz`

Also, see https://junegunn.github.io/fzf/search-syntax/#how-to-get-better-results

**Reply by simonschmidt** [reply, marked as answer]
> #3528

That's almost exactly what I wanted,  I just needed a small tweak to let me emit the query even if there are results:

```
export FZF_CTRL_R_OPTS='--bind "ctrl-g:print-query"'
```

Which let's me hit Ctrl+g at any point during the search.

Thanks for pointing me in the right direction with `FZF_CTRL_R_OPTS`!
